### ISSUE TITLE
`alt_tel` and `az_tel` outside of accepted window

### ISSUE THREAD

**Issue by vuillaut** [cta-observatory/cta-lstchain#282]
astropy transform coords requires angles between -pi and pi and fails in `reco_source_position_sky` with `alt_tel` and `az_tel` outside of this range (not sure of the unit)

**Comment by rlopezcoto** [comment]
Shouldn't this pop up in the same way also on `ctapipe`? How is it handled there?

**Comment by vuillaut** [comment]
The problem is that MC never give such values, so I don't think the problem appeared before.
The problem appears with the real pointing values of the telescope.

**Comment by rlopezcoto** [comment]
Actually I got these errors when analyzing muon MC with
`lstchain_mc_r0_to_dl1`, but I did not find them with before.
In any case, I agree we need to work it out.

**Comment by vuillaut** [comment]
> Actually I got these errors when analyzing muon MC with `lstchain_mc_r0_to_dl1`

Ha, interesting. Could you point exactly the bug you had please to help debug?

**Comment by morcuended** [comment]
I got this error too. I will check out whether alt az values make even sense. Could it be a problem of unit transformation from deg to rad?

**Comment by vuillaut** [comment]
> Could it be a problem of unit transformation from deg to rad?

It could be a unit issue coupled with transfo issues but I get values of (alt,az) in the thousands... so I wasn't sure what was the unit to start with. I see that the drive is giving angles in degrees. They should be converted in radians at reading I think to be aligned with the rest of lstchain and if possible assigned astropy units. Thanks for looking into this @morcuended !

**Comment by morcuended** [comment]
To keep track this is the error message I've been getting when running dl1->dl2 script:

```
Traceback (most recent call last):
  File "lstchain/scripts/lstchain_mc_dl1_to_dl2.py", line 89, in <module>
    main()
  File "lstchain/scripts/lstchain_mc_dl1_to_dl2.py", line 79, in main
    dl2 = dl1_to_dl2.apply_models(data, cls_gh, reg_energy, reg_disp_vector, custom_config=config)
  File "/fefs/home/daniel.morcuende/software/cta-lstchain/cta-lstchain/lstchain/reco/dl1_to_dl2.py", line 406, in apply_models
    az_tel * u.rad)
  File "/fefs/home/daniel.morcuende/software/cta-lstchain/cta-lstchain/lstchain/reco/utils.py", line 197, in reco_source_position_sky
    return camera_to_sky(src_x, src_y, focal_length, pointing_alt, pointing_az)
  File "/fefs/home/daniel.morcuende/software/cta-lstchain/cta-lstchain/lstchain/reco/utils.py", line 227, in camera_to_sky
    pointing_direction = SkyCoord(alt=pointing_alt, az=pointing_az, frame=horizon_frame)
  File "/home/daniel.morcuende/.conda/envs/mycta/lib/python3.7/site-packages/astropy/coordinates/sky_coordinate.py", line 257, in __init__
    frame_cls(**frame_kwargs), args, kwargs)
  File "/home/daniel.morcuende/.conda/envs/mycta/lib/python3.7/site-packages/astropy/coordinates/sky_coordinate_parsers.py", line 244, in _parse_coordinate_data
    valid_components.update(_get_representation_attrs(frame, units, kwargs))
  File "/home/daniel.morcuende/.conda/envs/mycta/lib/python3.7/site-packages/astropy/coordinates/sky_coordinate_parsers.py", line 589, in _get_representation_attrs
    valid_kwargs[frame_attr_name] = repr_attr_class(value, unit=unit)
  File "/home/daniel.morcuende/.conda/envs/mycta/lib/python3.7/site-packages/astropy/coordinates/angles.py", line 514, in __new__
    self._validate_angles()
  File "/home/daniel.morcuende/.conda/envs/mycta/lib/python3.7/site-packages/astropy/coordinates/angles.py", line 531, in _validate_angles
    'got {0}'.format(angles.to(u.degree)))
ValueError: Latitude angle(s) must be within -90 deg <= angle <= 90 deg, got [3405.95007773 3405.95008611 3405.95013526 ... 3407.20390473 3407.20394044
 3407.20394991] deg
```

I first checked altitude and azimuth coordinates values stored in the recent DL1 files. All of them are within the usual limits: altitude < 90 deg, azimuth < 360 deg.

As @vuillaut suggested I assigned astropy units, first in `lstchain/pointing/pointings.py`:
```
ev_azimuth = np.interp(ev_time, run_times, run_azimuth) * u.deg                                             
ev_altitude = np.interp(ev_time, run_times, run_altitude) * u.deg
```
and then transform them to radians before being stored since they are used in this unit later on:

```
if pointing_file_path and event_timestamps > 0:                                                                                                                                             
    azimuth, altitude = pointings.cal_pointingposition(event_timestamps, drive_data)                                                                                                        
    event.pointing[telescope_id].azimuth = azimuth.to(u.rad)                                                                                                                                
    event.pointing[telescope_id].altitude = altitude.to(u.rad)                                                                                                                              
    dl1_container.az_tel = azimuth.to(u.rad).value                                                                                                                                          
    dl1_container.alt_tel = altitude.to(u.rad).value
```
The first error does not appear anymore. DL1 to DL2 script works now fine as it is. Should I make a PR with these changes?

**Comment by vuillaut** [comment]
Hi @morcuended 
Thanks for tracking this bug!

> I first checked altitude and azimuth coordinates values stored in the recent DL1 files. All of them are within the usual limits: altitude < 90 deg, azimuth < 360 deg.

Isn't this contradictory with the error you show? or we are not talking about the same altitude/azimuth?

> ```ev_azimuth = np.interp(ev_time, run_times, run_azimuth) * u.deg```

Indeed, if `ev_azimuth` were in degrees but without unit, they were probably dealt with as if they are in radians later on, hence the obtained angles.

>  ```dl1_container.az_tel = azimuth.to(u.rad).value ```                                                                                                                                       

you can just set `dl1_container.az_tel = azimuth` since the unit is already given in the definition of `dl1_container.az_tel`

> DL1 to DL2 script works now fine as it is. Should I make a PR with these changes?

absolutely, thank you very much.

**Comment by rlopezcoto** [comment]
Solved in #283
